gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -141,16 +141,16 @@ public BatcherImpl(
     // to avoid deadlocking
     if (flowController.getLimitExceededBehavior() != LimitExceededBehavior.Ignore) {
       Preconditions.checkArgument(
-          flowController.getMaxOutstandingElementCount() == null
+          flowController.getMaxElementCountLimit() == null
```

Comment:
Looks like these methods got renamed since the `part 1` PR. Just curious why?

Reply:
In FlowController the semaphore is named "outstandingElementCount", and the limit setting was "maxOutstandingElementCount". The new variable I added for dynamic flow control settings is "currentOutstandingElementCount", which is a bit confusing to read, so renaming them to specify this is the limit. I think it's a bit clearer now? But let me know if you prefer to keep the original namings or have some better ideas :)